### PR TITLE
Catch errors that occur during fetch job and update FetchMonth status.

### LINKED ISSUE #137: Report failure when Fetch Job fails
In a recent set of fetch jobs, `/was_unaccessioned_data` ran out of disk space.

Actual result:

The Fetch Months displayed that they were running.

The jobs were requeued:
Screen Shot 2019-09-03 at 9 36 54 AM

And HB was notified:
Screen Shot 2019-09-03 at 9 37 34 AM

Expected result:
All of the above, except the status of the fetch month should be failure and the failure reason should be provided.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #137

### CODE REVIEW

File: spec/jobs/fetch_job_spec.rb [hunk 1]
```diff
@@ -7,14 +7,16 @@
   let(:fetch_month) { create(:fetch_month, collection: collection) }
 
   describe '#perform_now' do
-    before do
-      expect(Open3).to receive(:capture3).and_return([nil, stderr, status])
-      expect(FileUtils).to receive(:mkdir_p).with('tmp/jobs/AIT_915/2017_11')
-    end
+    # before do
```
Comment (jcoyne) [inline spec/jobs/fetch_job_spec.rb]: Cruft?

File: spec/jobs/fetch_job_spec.rb [hunk 2]
```diff
@@ -57,12 +61,33 @@
...
+
       it 'raises an error successfully' do
         expect { described_class.perform_now(fetch_month) }.to raise_error 'Fetching WARCs failed: Ooops'
         expect(fetch_month.status).to eq 'failure'
         expect(fetch_month.failure_reason).to eq 'Fetching WARCs failed: Ooops'
       end
     end
+
+    context 'when a non-fetch error occurs' do
+      # let(:status) { instance_double(Process::Status, success?: false) }
```
Comment (jcoyne) [inline spec/jobs/fetch_job_spec.rb]: Cruft

File: spec/jobs/fetch_job_spec.rb [hunk 3]
```diff
@@ -57,12 +61,33 @@
...
+      # let(:stderr) { 'Ooops' }
+
+      before do
+        expect(FileUtils).to receive(:mkdir_p).with('tmp/jobs/AIT_915/2017_11').and_raise(Errno::ENOSPC)
+      end
+
+      it 'raises an error successfully' do
+        # expect(Open3).to receive(:capture3).and_return([nil, stderr, status])
```
Comment (jcoyne) [inline spec/jobs/fetch_job_spec.rb]: cruft?

File: spec/jobs/fetch_job_spec.rb (hunk 1 above)
Comment (justinlittman) [reply spec/jobs/fetch_job_spec.rb]: Geez ... I need a cruft check pre-commit hook.

**Pull request merged**